### PR TITLE
PLAT-1791: Added disableBeforeDate and disableAfterDate props to zen-date-picker

### PR DESCRIPTION
**Pull request by f-jauregui**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1791)

#### Description
As per JIRA ticket

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

**Review by homer0 — COMMENTED** [review]
Nice 🤘... but I have a couple of comments/questions:

- `disableAfterDate` also disables "the date": If I tell it to disable the dates after `04/22/2021` it will not only disable `23` and on, but also `22`; not sure what the rest thinks, but I think it could be confusing if it says "after" but it also disables that date. Also, the same thing doesn't happen with `disableBeforeDate`, it disables all dates but not "the date".
- I saw that you did the conversion on the `componentDidLoad`; if I'm not wrong, that works for a React based application, but if the attribute were to change without creating a new instance, wouldn't that mean that the value wouldn't be parsed again?
- I think we should probably say somewhere that the `disableBeforeDate` and `disableAfterDate` values should be compatible with the `Date` constructor. Also, not sure if the data type shouldn't be `Date` too, like `value`... otherwise the implementation will need to manage the same component with two different "APIs": JS for `value` and HTML for `disable...`; what do you think?
- When calculating the disabled dates, you store them on `beforeDate` and `afterDate`: I think we should at least add some inline doc about those, because without the `disable` prefix, reading them without context doesn't tell you anything about what they represent. Also, shouldn't they be `private`?

Sorry :P; if you prefer, we can talk about it offline.

File: src/components/zen-date-picker/zen-date-picker.tsx [hunk 1]
```diff
@@ -221,6 +229,14 @@ export class ZenDatePicker {
     } else {
       this.dateChanged(this.value); // set today date
     }
+
+    if (this.disableBeforeDate !== 'null' && this.disableBeforeDate !== null) {
+      this.beforeDate = isValid(new Date(this.disableBeforeDate)) ? new Date(this.disableBeforeDate) : null;
```
Comment (chavlji) [inline src/components/zen-date-picker/zen-date-picker.tsx:234]: refactor: put `new Date(this.disableBeforeDate)` into var
same for `new Date(this.disableAfterDate)`

**Review by chavlji — CHANGES_REQUESTED** [review]
Works as expected! 👍🏻 

One minor thing: I think we should check if date is valid (between before and after) in the `value` and `formattedDate` watcher. If it's not, we should set it to nearest available date.
Eg. if I set `disableAfter` to `04/15/2021` and value to `04/25/2021`, value should be `04/15/2021`

**Review by chavlji — COMMENTED** [review]
Works better, but I get this error: date is set to `disabledAfter` + 1 day. It should be to `disabledAfter` exactly:

https://user-images.githubusercontent.com/5729421/115710513-b8fb2c80-a372-11eb-8689-dba30b3725b3.mov

**Comment by f-jauregui** [comment]
> Works better, but I get this error: date is set to `disabledAfter` + 1 day. It should be to `disabledAfter` exactly:
> 
>  Screen.Recording.2021-04-22.at.13.55.46.mov

Fixed. @chavlji, please review again! 😄

**Pull request merged**